graphql/resolvers/userDataCrud.ts

```diff
@@ -31,3 +36,38 @@ export const updateUserNames = withUserContainer<
 
   return user
 })
+
+export const updateUserPassword = withUserContainer<
+  Promise<SuccessResponse>,
+  MutationUpdateUserPasswordArgs
+>(async (_, args, { req }) => {
+  const user = req.user!
+  const { currentPassword, newPassword, newPasswordAgain } = args
+
+  if (newPassword !== newPasswordAgain) {
+    throw new Error("Passwords don't match")
+  }
+
+  const userFromDB = await prisma.user.findFirst({ where: { id: user.id } })
+
+  const validLogin = userFromDB!.password
+    ? await bcrypt.compare(currentPassword, userFromDB?.password!)
+    : false
```

Comment:
```ts
const validLogin = userFromDB
    ? await bcrypt.compare(currentPassword, userFromDB.password!)
    : false
```
I'd suggest writing here like this, makes the flow clearer and reduces extra assertions
or even 
```ts
const validLogin = userFromDB
    ? await bcrypt.compare(currentPassword, userFromDB.password as string)
    : false
```

Reply:
I'll update the code. Thanks for the comments!